encodings/dict/src/dict.rs

```diff
@@ -51,7 +52,7 @@ impl DictArray {
     #[inline]
     pub fn codes(&self) -> Array {
         self.as_ref()
-            .child(1, &self.metadata().codes_dtype, self.len())
+            .child(1, &DType::from(self.metadata().codes_ptype), self.len())
```

Comment:
Prefer `.into()` for consistency with the other cases?

Reply:
Consistency is good (plus PType -> DType conversion is quite cheap), but FWIW I would vote that we prefer `from` over `into` for future code.

Reply:
Yeah, I much prefer `from` over `into`, more readable